pkg/controller/infinispan/client.go

```diff
@@ -12,12 +12,23 @@ import (
 	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
 )
 
+// Create a cluster for connecting to pods created on the 2.0.x channel
+func NewLegacyCluster(i *v1.Infinispan, kubernetes *kube.Kubernetes) (*ispn.Cluster, error) {
+	// Use the user secret in order to obtain the old credentials used by the legacy server
+	pass, err := users.AdminPassword(i.GetSecretName(), i.Namespace, kubernetes)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve identities when creating legacy Cluster instance: %w", err)
+	}
+	return ispn.NewCluster(consts.DefaultOperatorUser, pass, i.Namespace, i.GetEndpointScheme(), 11222, kubernetes), nil
```

Comment:
We have constant for the 11222 port

Reply:
I deliberately avoided using the constant because any future updates to this value will result in the legacy behaviour being broken.

Reply:
Probably we can introduce another constant for better visibility

Reply:
The code is only used in one place, so unless we need it elsewhere I think a constant is overkill.